### PR TITLE
feat: SSE / real time flags support #32 

### PR DESCRIPTION
**Pull request by gazreese**
## Description

This PR implements the Real Time Flags feature #32 and part of https://github.com/Flagsmith/flagsmith/issues/1498 for iOS.

The functionality is implemented using the Kotlin SDK's implementation as a working example, as this is the platform most closely aligned with the iOS implementation.

One difference of note is that the iOS SDK does not make use of any HTTP client library, using URLSession directly. As such its SSE implementation contains some lower-level functionality not found in the other implementations such as more granular error handling and increasing delay timer on most error conditions.

It also felt prudent to update the Example app to support SwiftUI and in-turn provide a reactive view of the current flags based on the new `var flagStream: AsyncStream<[Flag]>` provided by the SDK. This allowed for 'real life' manual testing through various network conditions to verify the integrity of the SSE connection.

Pre-iOS 13 compatibility has been maintained as was done in the current SDK, though the need for this is probably up for discussion. Validating the SDK on iOS 12.x wasn't possible on my Mac. The implementation might have been a little tidier it this wasn't the case, and obvious improvements are commented in the code.

Unit tests have been added for all core business logic where possible without butchering the implementation. Significant changes would be required to get any more coverage really.

- [ ] Documentation will be required when the PR is approved.

## Regression Test Recommendations

I've tested this on Swift 5.10 and the new Xcode with Swift 6 and everything was fine. Xcode is really good at telling you about deprecations and compatibility issues and there are no warnings.

Has all been linted and formatted.

Would be nice to try this on iOS 12.x if possible.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: FlagsmithClient/Classes/Internal/APIManager.swift
Comment (matthewelwell) [inline FlagsmithClient/Classes/Internal/APIManager.swift:203]: And here, can we use something other than print?

File: FlagsmithClient/Classes/Internal/SSEManager.swift
Comment (matthewelwell) [inline FlagsmithClient/Classes/Internal/SSEManager.swift:54]: Probably we mean `environment` here instead of organisation. Also note that we defiantly use British english throughout the Flagsmith project ;)

File: FlagsmithClient/Classes/Internal/SSEManager.swift
Comment (matthewelwell) [inline FlagsmithClient/Classes/Internal/SSEManager.swift:87]: This is mostly just my pythonic brain thinking that defining a variable called `json` is a no-no, but regardless, I think we can be more explicit here

```suggestion
            let jsonMessageData = line.replacingOccurrences(of: "data:", with: "")
```

File: FlagsmithClient/Tests/SSEManagerTests.swift
Comment (matthewelwell) [inline FlagsmithClient/Tests/SSEManagerTests.swift:72]:
```suggestion
    func testInvalidSSEDataNotANum() {
```

**Review by matthewelwell — COMMENTED** [review]
Looks good overall, but I've added a few minor comments.

File: FlagsmithClient/Classes/Flagsmith.swift
Comment (gazreese) [reply FlagsmithClient/Classes/Flagsmith.swift:360]: There is, but it's more for Swift on server - https://github.com/apple/swift-log.

Most packages seem to just use print(). I think these error ones will be pretty handy if we need to support a customer but happy to remove or switch, up to you @matthewelwell . We had print() in there before.

File: Example/Podfile.lock
Comment (gazreese) [reply Example/Podfile.lock:2]: It's easier to update this before I merge @matthewelwell so please have a think and let me know

File: FlagsmithClient/Classes/Flagsmith.swift
Comment (gazreese) [reply FlagsmithClient/Classes/Flagsmith.swift:360]: Just checked again and there's already a lot of use of `print()` @matthewelwell . Don't forget, if we pull a dependency in we're passing that on to our customers in app size and / or version restrictions, I think we're best off without too much in the way of dependencies.

File: FlagsmithClient/Classes/Internal/SSEManager.swift
Comment (gazreese) [reply FlagsmithClient/Classes/Internal/SSEManager.swift:54]: Have updated throughout

File: Example/Podfile.lock
Comment (matthewelwell) [reply Example/Podfile.lock:2]: We'll want to bump the minor version for this change, but we also have another change (which also warrants a minor version) waiting for release [here](https://github.com/Flagsmith/flagsmith-ios-client/commit/ba0fb2bc930043e8a867777b6a4034618f2b8799). So I'd recommend just pulling the version change out of this PR and we'll submit a separate version bump PR after this is merged.

File: FlagsmithClient/Classes/Flagsmith.swift
Comment (matthewelwell) [reply FlagsmithClient/Classes/Flagsmith.swift:360]: Understood, let's leave it then.

**Review by matthewelwell — APPROVED** [review]
Approved, but let's remove the version bump.

File: Example/Podfile.lock
Comment (gazreese) [reply Example/Podfile.lock:2]: OK great, would be handy to put a release flow in that updates these automatically, commits the changes and makes a new release on GitHub.

Comment (gazreese) [reply Example/Podfile.lock:2]: Hi @matthewelwell this is just the lock file for the example app, which will pull in whatever the version was set to when the last person used it.

I think we can safely remove and ignore this file as it's clearly confusing with PRs.

Main library is a different file and is 3.7.0

**Comment by gazreese** [comment]
> Approved, but let's remove the version bump.

As above, there isn't actually a bump

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**
